src/Components/Admin/AssetModelViewer/assetModelViewer.js

```diff
@@ -0,0 +1,153 @@
+    },
+    drawer: {
+        width: drawerWidth + "vw",
+        flexShrink: 0,
+    },
+    drawerPaper: {
+        width: drawerWidth + "vw",
+    },
+    // necessary for content to be below navbar
+    toolbar: theme.mixins.toolbar,
+    content: {
+        flexGrow: 1,
+        backgroundColor: theme.palette.background.default,
+        padding: theme.spacing(3),
+    },
+    textfields: {
+        "& .MuiTextField-root": {
+            margin: theme.spacing(2),
+            width: "15vw",
+        },
+    },
+}));
+
+export default function AssetModelViewer({
+    match: {
+        params: { assetId },
+    },
+}) {
+    const classes = useStyles();
+    const handleError = useErrorHandler();
+    const [asset, setAsset] = useState({});
+    const [name, setName] = useState("");
+    const [objectType, setObjectType] = useState("");
+    const [buttonText, setButtonText] = useState("Save");
+    const [modelGLB, setModelGLB] = useState("");
+
+    useEffect(() => {
+        const getObjectAsset = async () => {
+            const data = await getAsset(assetId, handleError);
+            setName(data.name);
+            setObjectType(data.obj_type);
+            // TODO: Replace with asset's s3_link once it contains a legit link
+            setModelGLB(jsonMock.linksGLB[0]);
+            setAsset(data);
+        };
+
+        if (assetId) {
+            getObjectAsset();
+        }
+    }, [assetId, handleError]);
+
+    const handleNameChange = (event) => {
+        setName(event.target.value);
+    };
+
+    const handleObjectTypeChange = (event) => {
+        setObjectType(event.target.value);
+    };
+
+    async function updateAsset(event) {
+        event.preventDefault();
+        setButtonText("Saved!");
+        setTimeout(() => {
+            setButtonText("Save");
+        }, 2000);
+        editAsset(assetId, name, objectType, asset.s3_key, handleError);
+    }
+
+    return (
+        <div className={classes.root}>
+            <CssBaseline />
+            <Drawer
+                className={classes.drawer}
+                variant="permanent"
+                classes={{
+                    paper: classes.drawerPaper,
+                }}
+                anchor="left"
+            >
+                <div className={classes.toolbar} />
+                <Divider />
+
+                <form
+                    className={classes.textfields}
+                    noValidate
+                    onSubmit={updateAsset}
+                    autoComplete="off"
+                >
+                    <Typography variant="h6" align="center">
+                        Asset Model Viewer
+                    </Typography>
+                    <TextField
+                        id="outlined-helperText"
+                        label="Asset Name"
+                        value={name}
+                        onChange={handleNameChange}
+                        variant="outlined"
+                    />
+
+                    <TextField
```

Comment:
A `Select` with options here would be a better option here imo, getting admins to select between `Background` and `Object` rather than having to know the exact thing. Because if they input `Object` instead of `object`, it will also error.

Reply:
^^ Agree with this

Reply:
Replaced textfield with selector